mne/io/base.py

```diff
@@ -1028,26 +1013,25 @@ def notch_filter(self, freqs, picks=None, filter_length='10s',
         -----
         For details, see :func:`mne.filter.notch_filter`.
         """
-        raw = self.copy() if copy else self
-        fs = float(raw.info['sfreq'])
+        fs = float(self.info['sfreq'])
         if picks is None:
-            picks = _pick_data_or_ica(raw.info)
+            picks = _pick_data_or_ica(self.info)
             # let's be safe.
             if len(picks) < 1:
                 raise RuntimeError('Could not find any valid channels for '
                                    'your Raw object. Please contact the '
                                    'MNE-Python developers.')
-        _check_preload(raw, 'raw.notch_filter')
-        raw._data = notch_filter(
-            raw._data, fs, freqs, filter_length=filter_length,
+        _check_preload(self, 'raw.notch_filter')
+        self._data = notch_filter(
+            self._data, fs, freqs, filter_length=filter_length,
             notch_widths=notch_widths, trans_bandwidth=trans_bandwidth,
             method=method, iir_params=iir_params, mt_bandwidth=mt_bandwidth,
             p_value=p_value, picks=picks, n_jobs=n_jobs, copy=False)
-        return raw
+        return self
 
     @verbose
     def resample(self, sfreq, npad=None, window='boxcar', stim_picks=None,
-                 n_jobs=1, events=None, copy=False, verbose=None):
+                 n_jobs=1, events=None, copy=None, verbose=None):
```

Comment:
If I'm not mistaken, the window param isn't used. I think it should be passed to ``resample`` at line 1134.